### DISCUSSION TITLE
Bookworm and ePub files

### DISCUSSION THREAD

**Discussion by Cambridgeport90** [mush42/bookworm#84] [General]
I have noticed that when different Epub files are opened with Bookworm, depending on their formatting, it seems that the entire book is somehow missing... it will skip over the entire content of the book unless each chapter or section is clearly marked out in the table of contents. Otherwise, nothing happens. I wanted to post a discussion in the event that this isn't a bug but just a weird quirk. running Windows 10 and the latest version of Bookworm beta; beta 0.4.

**Comment by mush42** [comment]
Hi @Cambridgeport90,

Unfortunately, This issue was a long standing issue, and it blocked some important features from being implemented.

But I'm glad to inform you that this issue has already been fixed in [this commit](https://github.com/blindpandas/bookworm/commit/e86ff4dfa9c5fe4287bedf26a98fabe7c9075ecb) currently in the **next** branch. Now we have a totally new EPUB parser that handles edge cases better than the old one.

Stay Safe,

**Reply by Cambridgeport90** [reply]
Is there a release built on that second branch that I can grab, or is that still in the works? I've been keeping an eye on this.

**Reply by mush42** [reply]
Hi,

Unfortunately, we are unable to provide builds for the branch **next**. The code is very much work in progress and unstable.

As soon as we move this to develop, we'll provide an alpha build for testing.

Regards,

**Comment by Cambridgeport90** [comment]
Good thing I subscribed to all activity up here, then.
